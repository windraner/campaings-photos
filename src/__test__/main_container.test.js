import React from 'react';
import MainContainer from '../components/main_container/main_container';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

const mock = {
  loading: false,
  isSelectedAll: false,
  isFilterShown: true,
  itemsView: 'list',
  page: 1,
  perPage: 25,
  pagesCount: 10,
  switchHandler() {},
  stateHandler() {},
};

describe('MainContainer', () => {
  it('matches the snapshot', () => {
    const tree = shallow(<MainContainer {...mock} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
  it('matches the snapshot', () => {
    const tree = shallow(<MainContainer {...mock} isFilterShown={false} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
