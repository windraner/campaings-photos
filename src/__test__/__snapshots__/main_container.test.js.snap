// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContainer matches the snapshot 1`] = `
<div
  className="campaigns-photos__main-container"
>
  <TopBar
    isFilterShown={true}
    isSelectedAll={false}
    itemsView="list"
    stateHandler={[Function]}
    switchHandler={[Function]}
  />
  <BottomBar
    page={1}
    pagesCount={10}
    perPage={25}
    stateHandler={[Function]}
  />
</div>
`;

exports[`MainContainer matches the snapshot 2`] = `
<div
  className="campaigns-photos__main-container"
>
  <TopBar
    isFilterShown={false}
    isSelectedAll={false}
    itemsView="list"
    stateHandler={[Function]}
    switchHandler={[Function]}
  />
  <BottomBar
    page={1}
    pagesCount={10}
    perPage={25}
    stateHandler={[Function]}
  />
</div>
`;
